draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,528 +1,448 @@
+                 | data             |      | data,            | sending
+                 |                  |      | requirements     | rate (or
+                 v                  |      v                  | window)
+  +-----------------------------+   |    +--------------------------+
+  |             FEC             |        | Transport (including CC) |
+  +-----------------------------+   |    +--------------------------+
+    ^                   |           |              ^
+    | information       | coded     |              | network
+    | about repaired    | data      |              | measurements
+    | ratio             -------------              |
+
+                     Figure 3: FEC above the transport
+
+   The coded packets are sent within what the congestion window allows.
+
+   TODO : ADD MORE TEXT
+
+5.  FEC within the transport
+
+   Figure 4 illustrates the FEC within the transport case.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 5]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+                          |
+                          | data, requirements
+                          |
+                          v
+           +---------------------+
+           | Transport           |
+           | +-----+  +-----+    | coded data
+           | | FEC |  | CC  |    |
+           | +-----+  +-----+    | ----->
+           |                     |
+           +---------------------+
+                   ^          ^
+                   |          |
+           information    network
+           about          measurements
+           repaired ratio
+
+                      Figure 4: FEC in the transport
+
+   The coded packets are sent within what the congestion window allows,
+   such as in [CTCP].  Examples of the solution would be sending coded
+   packets when there is no more data to transmit or preferably send
+   coded packets instead of the following packets in the send buffer.
+
+   TODO : ADD MORE TEXT
+
+6.  FEC below the transport
+
+   Figure 5 illustrates the FEC below the transport case.
+
+                  |                  --------                  ^
+                  | data             |      | data            | coded
+                  | requirements     |      |                 | data
+                  v                  |      v                 |
+   +-----------------------------+   |    +-------------------------+
+   | Transport (including CC)    |        |           FEC           |
+   +-----------------------------+   |    +-------------------------+
+     ^                   |           |              ^
+     | network           |           |              | information
+     | measurements      |           |              | about repair
+     |                   -------------              | ratio
+
+                     Figure 5: FEC below the transport
```

Comment:
Again, I would layer this, putting the FEC box below the transport box - and again, please change the caption to say "FEC below the transport (sender side)".

Proposal:

         |                    ^
         | data,              | sending
         | requirements       | rate (or
         v                    | window)
    +---------------------------------------+
    |       Transport (including CC)        |
    +---------------------------------------+
                    |                    ^
                    | data               | network
                    |                    | measurements
                    v                    |
    +-------------------------------+
    |              FEC              |
    +-------------------------------+
         |                    ^
         | coded              | information
         | data               | about repaired
         v                    | ratio

Reply:
ACK